### PR TITLE
[WIP] Bitpay rates api

### PR DESCRIPTION
**Pull request by Kukks**
This PR introduces a bitpay compatible rates api.
```
/rates
/rates/{currencyCode}
/rates/{baseCurrencyCode}/{currencyCode}
```

I've added a new property to `ISupportedPaymentMethod` so that a payment method may specify which currency code it handles. This way the `/rates` & `/rates/{currencyCode}`  endpoint will map all available payment methods to available currency pairs to return.
E.g. a btcpayserver instance with btc and ltc configured would return pairs of "BTC_LTC" and "LTC_BTC" under `/rates` and "BTC_LTC" under `/rates/btc`.

`rates/x/y`  would map directly to the specified values.  

The only incompatibility left is that we still need to specify the store id but maybe that is fine and should be handled from the clients directly (see https://github.com/btcpayserver/btcpayserver/pull/197 )

### CODE REVIEW

**Comment by NicolasDorier** [comment]
I updated the NBitpayClient package, can you add some tests?

**Comment by Kukks** [comment]
Will do

**Comment by Kukks** [comment]
Trying to write some UTs but the test execution is taking forever. Does it usually take >10mins to execute?
image

**Comment by NicolasDorier** [comment]
mmh this is tricky. Unlike Bitpay, we have no way to enumerate all available rates from a cryptoCode.

**Comment by NicolasDorier** [comment]
I am unsure if there is a nice way to get it like Bitpay. The root of the problem is that we have no generic way to enumerate the rates.

**Comment by Kukks** [comment]
@NicolasDorier I'm going to go ahead and get the final touches on this done. Do you have any requirements to add?

**Comment by NicolasDorier** [comment]
it seems good to me as long as it does not break stuff, ready to merge?

**Comment by Kukks** [comment]
ok we can merge it in @NicolasDorier 

So to summarize, 
This adds in the bitpay compliant rates api with one variation to the original api: the request must have a storeid query paramater or the request must be authenticated.

I will add open up a PR to NBitPayClient to send those requests as authenticated.

**Comment by Kukks** [comment]
Actually give me a min, need to verify some stuff

**Comment by Kukks** [comment]
Added in some small changes and done from my end.

File: BTCPayServer/Payments/ISupportedPaymentMethod.cs [hunk 1]
```diff
@@ -13,6 +13,7 @@ namespace BTCPayServer.Payments
     /// </summary>
     public interface ISupportedPaymentMethod
     {
+        string CryptoCode { get;}
```
Comment (NicolasDorier) [inline BTCPayServer/Payments/ISupportedPaymentMethod.cs]: Why? you can use `PaymentId.CryptoCode`.

Comment (Kukks) [reply BTCPayServer/Payments/ISupportedPaymentMethod.cs]: Ah, did not know exactly what PaymentId was until now, removed it now

**Pull request merged**